src/codeGenerator/generateBodyCode.ts

```diff
@@ -347,5 +347,17 @@ const getConditionStatement = ({
         return `typeof ${getName(nameStack, namePrefix)} !== '${astNode.type}'`;
     }
 
-    return `${getName(nameStack, namePrefix)} !== ${astNode.type}`;
+    /**
+     * This function returns the most appropriate condition statement,
+     * therefore for those types it returns pure boolean value.
+     *
+     * However the generated whole conditional expression would be useless.
+     */
+    const untyped = ['any', 'unknown'];
+    if (untyped.includes(astNode.type)) {
+        return `false`;
+    }
+
+    /** reject by default */
+    return `true`;
```

Comment:
@drowsy-probius 
I Didn't check for the test case but this may lead to crash for union types :(
like `union: 'a' | 'b' | 'c'`

I will test for this later